ElementX/Sources/Services/Client/ClientProxy.swift

```diff
@@ -162,34 +169,20 @@ class ClientProxy: ClientProxyProtocol {
         }
     }
 
-    func loadUserAvatar(policy: AvatarLoadingPolicy) async {
-        if policy != .skipCache {
-            await Task.dispatch(on: clientQueue) {
-                guard let urlString = try? self.client.cachedAvatarUrl() else {
-                    return
-                }
-                self.avatarUrlSubject.value = URL(string: urlString)
-            }
-        }
-        if policy != .cacheOnly {
-            await Task.dispatch(on: clientQueue) {
-                do {
-                    if let urlString = try self.client.avatarUrl() {
-                        self.avatarUrlSubject.value = URL(string: urlString)
-                    } else {
-                        self.avatarUrlSubject.value = nil
-                    }
-                } catch {
-                    return
+    func loadUserAvatar() async {
+        await Task.dispatch(on: clientQueue) {
+            do {
+                if let urlString = try self.client.avatarUrl() {
+                    self.avatarURLSubject.value = URL(string: urlString)
+                } else {
+                    self.avatarURLSubject.value = nil
                 }
```

Comment:
A nice little improvement here would be to
```suggestion
                let urlString = try self.client.avatarUrl()
                self.avatarURLSubject.value = urlString.flatMap(URL.init)
```